breakerbox-turbine/src/main/java/com/yammer/breakerbox/turbine/client/MarathonClient.java

```diff
@@ -0,0 +1,28 @@
+package com.yammer.breakerbox.turbine.client;
+
+import com.yammer.breakerbox.turbine.config.MarathonClientConfiguration;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.WebTarget;
+import javax.ws.rs.core.Response;
+
+/**
+ * Created by supreeth.vp on 23/05/17.
+ */
+public class MarathonClient {
+
+    private MarathonClientConfiguration marathonClientConfiguration;
+
+    public MarathonClient(MarathonClientConfiguration marathonClientConfiguration){
+        this.marathonClientConfiguration = marathonClientConfiguration;
+    }
+
+     public Response getServiceInstanceDetails() {
+            Client client = ClientBuilder.newClient();
```

Comment:
We reconstruct these Clients on every instance discovery run. Since the marathon client configurations don't modify ever after the initial construction of the InstanceDiscovery implementation. Can we just maintain a list of pre-constructed marathon clients instead?